src/core/MOM_barotropic.F90

```diff
@@ -4885,9 +4926,11 @@ subroutine barotropic_init(u, v, h, eta, Time, G, GV, US, param_file, diag, CS,
   ALLOC_(CS%IdyCv(CS%isdw:CS%iedw,CS%jsdw-1:CS%jedw)) ; CS%IdyCv(:,:) = 0.0
   ALLOC_(CS%dy_Cu(CS%isdw-1:CS%iedw,CS%jsdw:CS%jedw)) ; CS%dy_Cu(:,:) = 0.0
   ALLOC_(CS%dx_Cv(CS%isdw:CS%iedw,CS%jsdw-1:CS%jedw)) ; CS%dx_Cv(:,:) = 0.0
+  allocate(CS%IareaT_OBCmask(isd:ied,jsd:jed))
```

Comment:
Please change this line to `allocate(CS%IareaT_OBCmask(isd:ied,jsd:jed), source=0.0)` so that the halo points are initialized to 0 in non-reentrant domains.